spec/spec_helper.rb

```diff
@@ -4,12 +4,12 @@
 include Mercury::TestUtils
 
 # the block must return a Cps
-def test_with_mercury_cps(sources, queues, **kws, &block)
+def test_with_mercury_cps(sources, queues, **kws)
   em do
     seql do
       let(:m)  { Mercury::Monadic.open(**kws) }
       and_then { delete_sources_and_queues_cps(sources, queues) }
-      and_then { block.call(m) }
+      and_then { yield(m) }
```

Comment:
yep, you shouldn't _have to_ read the implementation of a function to know how to expect the function to behave.